### DISCUSSION TITLE
Start, shutdown and restart

### DISCUSSION THREAD

**Discussion by Koxx3** [Hieromon/AutoConnect#333] [Q&A]
Hello,

I am trying to start/stop/restart the portal but after the first stop, I can't start it again.
I looked at the examples and the documentation but I couldn't find the asnwer.

In delcarations, I have :
```
WebServerClass server;
AutoConnect portal(server);
AutoConnectConfig config;

```
In 'setup' (called in main setup), I have :
```
  // Responder of root page handled directly from WebServer class.
  server.on("/_ac", []() {...
  portal.config(config);
  server.stop();
```

In 'start', only called when I want to start : 
```
  server.begin();
  portal.begin();
```

In 'stop' only called when I want to start : 
```
  portal.end();
  server.stop();
```

The current code is crashing when calling 'start'.

Could you help to get a start / stop working please ?

when stopped, all wifi activities needs to be stoped.

**Comment by Koxx3** [comment]
in fact, even with "stop", it seems locked in 
`  portal.handleClient();`

**Comment by Hieromon** [comment]
You can terminate AutoConnect with the [`AutoConnect::end`](https://hieromon.github.io/AutoConnect/api.html#end) function. But it releases all resources such as WebServer instances that AutoConnect manages internally. The AutoConnect::handleClient function doesn't work as a matter of course.

Although not highly recommended, you should be able to run a web server stop with `WebServer::stop` only to stop and restart the portal. (However, I haven't tried this way. It is difficult to identify what happens in the WiFiServer library, and I think it will result in unstable behavior.)

Also,
> when stopped, all wifi activities needs to be stoped.

Change the WiFi mode to STA and call `WiFi.disconnect(true)`, then ESP32's WiFi will stop.

**Reply by Koxx3** [reply]
thanks. I'll try those solutions.
